source/jormungandr/jormungandr/scenarios/helper_classes/streetnetwork_path.py

```diff
@@ -80,27 +93,105 @@ def __init__(
         self._request_id = request_id
         self._async_request()
 
-    @new_relic.distributedEvent("direct_path", "street_network")
-    def _direct_path_with_fp(self):
-        with timed_logger(self._logger, 'direct_path_calling_external_service', self._request_id):
-            try:
-                return self._streetnetwork_service.direct_path_with_fp(
+    def poi_to_pt_object(self, poi):
+        return type_pb2.PtObject(poi=poi, uri=poi.uri, embedded_type=type_pb2.POI, name=poi.name)
+
+    def get_pt_objects(self, entry_point):
+        if include_poi_access_points(self._request, entry_point, self._mode):
+            for o in entry_point.poi.children:
+                yield self.poi_to_pt_object(o)
+        yield entry_point
+
+    def make_poi_access_points(self, fallback_type, best_dp_element):
+        entry_point = (
+            self._orig_obj if fallback_type == StreetNetworkPathType.BEGINNING_FALLBACK else self._dest_obj
+        )
+        if not include_poi_access_points(self._request, entry_point, self._mode):
+            return
+        via_poi_access = (
+            best_dp_element.origin
+            if fallback_type == StreetNetworkPathType.BEGINNING_FALLBACK
+            else best_dp_element.destination
+        )
+        _add_poi_access_point(fallback_type, via_poi_access, best_dp_element.response.journeys[0].sections)
```

Comment:
watch out the subscript of journeys... there could be **more than one journey** in a direct_path, for example, a direct_path of bike could return 3 journeys... 

from now on, we should always consider that a direct_path request may generate more than one journey... 

```suggestion
        for journey in best_dp_element.response.journeys:
            _add_poi_access_point(fallback_type, via_poi_access, journey.sections)
```

Reply:
Done